validator/sawtooth_validator/journal/publisher.py

```diff
@@ -370,7 +370,8 @@ def _finalize_block(self, block):
                 else:
                     block.add_batch(batch)
                     self._committed_txn_cache.add_batch(batch)
-                state_hash = result.state_hash
+                if result.state_hash is not None:
+                    state_hash = result.state_hash
```

Comment:
I tend to agree. From talking with @vaporos and @jsmitchell it sounds like there might be some use in the future for a multi-block scheduler, and so it wouldn't be just one state hash returned. Since the SerialScheduler is more of a first pass until we get the parallel scheduler implemented (and they should be entirely interchangable), I don't think we should try to change the interface for the scheduler.